### ISSUE TITLE
[BUG][SDW][0x8]"rt715 sdw:3:25d:715:0: Initialization not complete, timed out" occurred during multipipline stress test

### ISSUE THREAD

**Issue by YvonneYang2** [thesofproject/linux#1660]
**Describe the bug**
Split this issue from #1637 , on options snd-sof-intel-hda-common sdw_clock_stop_quirks=0x8,
"rt715 sdw:3:25d:715:0: Initialization not complete, timed out" occurred when do multipipline stress test with kernel as below:

**Kernel 1.** Latest integration/soundwire-latest branch(121c751) + #1640
**Kernel 2.** Latest https://github.com/plbossart/sound/tree/fix/soundwire-interrupt-hardening (top commit: 88d8c0d)

Both CML-H and CML-U SDW laptop can reproduce the issue. 

**To Reproduce**
1. Boot up with options 0x8
2. Run with command "./multiplex_pipeline-test_update.sh pcm0p pcm2p pcm5p pcm1c pcm4c"
3. Check logs in folder named logs.
[multiplex_pipeline-test_update.sh.txt](https://github.com/thesofproject/linux/files/4025499/multiplex_pipeline-test_update.sh.txt)

**Reproduced rate**
On **CML-U** Laptop DVT2 with ALC711+ ALC1308+ ALC715 in SDW mode
With kernel 1: Run 300 times, [failed_multipipeline_test_136_dmesg.log](https://github.com/thesofproject/linux/files/4025502/failed_multipipeline_test_136_dmesg.log)
With kernel 2: Run 300 times, [failed_multipipeline_test_26_dmesg.log](https://github.com/thesofproject/linux/files/4025504/failed_multipipeline_test_26_dmesg.log)

 On **CML-H** Laptop DVT1 with ALC711+ ALC1308+ ALC715 in SDW mode
With kernel 1: Run 300 times, on RT715 [failed_multipipeline_test_19_dmesg.log](https://github.com/thesofproject/linux/files/4029201/failed_multipipeline_test_19_dmesg.log)

With kernel 2: Run 300 times, on RT715 [failed_multipipeline_test_73_dmesg.log](https://github.com/thesofproject/linux/files/4029192/failed_multipipeline_test_73_dmesg.log)

**Expected result**
Multipipline stress test works well.

**Actual result**
"rt715 sdw:3:25d:715:0: Initialization not complete, timed out" or "rt711 sdw:0:25d....." occurred,  arecord0,4(rrt715) or arecord0,1(rt711) cannot work anymore until reboot.

<details>
 <summary>Dmesg</summary>

````
[ 9546.702949] intel-sdw sdw-master-3: sdw_cdns_clock_restart: start
[ 9546.704310] intel-sdw sdw-master-3: sdw_cdns_clock_restart: end
[ 9546.704311] intel-sdw sdw-master-3: intel_resume_runtime done
[ 9547.189528] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 9547.191053] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 9547.689502] sof-audio-pci 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 9547.691811] sof-audio-pci 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 9548.715926] rt715 sdw:3:25d:715:0: Initialization not complete, timed out
[ 9548.715942] intel-sdw sdw-master-3: intel_startup: SDW3 Pin2: start
[ 9548.715952] intel-sdw sdw-master-3: intel_startup: SDW3 Pin2: done
[ 9548.715961]  DMIC: ASoC: open FE DMIC
[ 9548.715970] sof-audio-pci 0000:00:1f.3: pcm: open stream 4 dir 1
[ 9548.715974] sof-audio-pci 0000:00:1f.3: period min 192 max 16384 bytes
[ 9548.715977] sof-audio-pci 0000:00:1f.3: period count 2 max 16
[ 9548.715979] sof-audio-pci 0000:00:1f.3: buffer max 65536 bytes
[ 9548.716064]  SDW0-Playback: ASoC: hw_params BE SDW0-Playback
````
</details>

**Comment by YvonneYang2** [comment]
Tested it on CML-U and CML-H sdw laptop with latest soundwire-latest branch, issue cannot be reproduced. So it seems issue is only reproduced with #1640 or on  https://github.com/plbossart/sound/tree/fix/soundwire-interrupt-hardening (88d8c0d)

**Comment by plbossart** [comment]
Sorry @YvonneYang2 I don't really buy these results. It's not really possible that the fixes for interrupts add more problems...
Do you mind re-trying both plain vanilla soundwire-latest and 88d8c0d?

**Comment by RanderWang** [comment]
please check https://github.com/thesofproject/linux/pull/1640#discussion_r363593191. Can you test https://github.com/thesofproject/linux/commit/88d8c0db8595fd3349e50e831a39d81a0a249d55 only ?

**Comment by YvonneYang2** [comment]
@plbossart 
Double checked it 1000 times with both plain vanilla soundwire-latest and 88d8c0d , issue cannot be reproduced. 
Current issue #1660 only occurred with PR#1640.

**Comment by YvonneYang2** [comment]
Close the issue since PR1640 is closed and replaced by #1665 . Checked it with soundwire-latest + #1665  , issue cannot be reproduced.
